### PR TITLE
FKM-9. Resolve problem with lost file counts.

### PR DESCRIPTION
**Pull request by dporplenko**

### CODE REVIEW

**Comment by dporplenko** [comment]
test this please

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/37/

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/38/

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/39/

**Comment by dporplenko** [comment]
test this please

**Comment by dporplenko** [comment]
test this please

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/40/

**Build Log**
*last 25 lines*

 ```
[...truncated 134 lines...]
skipping: [localhost]

TASK: [Removing possible abandoned dump] ************************************** 
skipping: [localhost]

TASK: [Importing from remote backup] ****************************************** 
skipping: [localhost]

TASK: [Copy latest backup to build folder] ************************************ 
changed: [localhost]

TASK: [Importing from local backup] ******************************************* 
failed: [localhost] => {"failed": true}
msg: ERROR 1114 (HY000) at line 17396: The table 'watchdog' is full


FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=20   changed=13   unreachable=0    failed=1   

Build step 'Execute shell' marked build as failure
```

**Comment by dporplenko** [comment]
test this please

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/41/

**Build Log**
*last 25 lines*

 ```
[...truncated 134 lines...]
skipping: [localhost]

TASK: [Removing possible abandoned dump] ************************************** 
skipping: [localhost]

TASK: [Importing from remote backup] ****************************************** 
skipping: [localhost]

TASK: [Copy latest backup to build folder] ************************************ 
changed: [localhost]

TASK: [Importing from local backup] ******************************************* 
failed: [localhost] => {"failed": true}
msg: ERROR 1114 (HY000) at line 10953: The table 'field_revision_field_image' is full


FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=20   changed=13   unreachable=0    failed=1   

Build step 'Execute shell' marked build as failure
```

**Comment by dporplenko** [comment]
test this please

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/42/

**Build Log**
*last 25 lines*

 ```
[...truncated 103 lines...]
	from /usr/lib/ruby/vendor_ruby/bundler/definition.rb:109:in `resolve_remotely!'
	from /usr/lib/ruby/vendor_ruby/bundler/installer.rb:83:in `run'
	from /usr/lib/ruby/vendor_ruby/bundler/installer.rb:14:in `install'
	from /usr/lib/ruby/vendor_ruby/bundler/cli.rb:247:in `install'
	from /usr/lib/ruby/vendor_ruby/thor/command.rb:27:in `run'
	from /usr/lib/ruby/vendor_ruby/thor/invocation.rb:121:in `invoke_command'
	from /usr/lib/ruby/vendor_ruby/thor.rb:363:in `dispatch'
	from /usr/lib/ruby/vendor_ruby/thor/base.rb:440:in `start'
	from /usr/bin/bundle:20:in `block in <main>'
	from /usr/lib/ruby/vendor_ruby/bundler/friendly_errors.rb:3:in `with_friendly_errors'
	from /usr/bin/bundle:20:in `<main>'
stdout: Fetching gem metadata from https://rubygems.org/.
Fetching source index from https://rubygems.org/
Unfortunately, a fatal error has occurred. Please see the Bundler
troubleshooting documentation at http://bit.ly/bundler-issues. Thanks!

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=9    changed=5    unreachable=0    failed=1   

Build step 'Execute shell' marked build as failure
```

**Comment by dporplenko** [comment]
test this please

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/43/

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/45/

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/46/

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/47/

**Build Log**
*last 25 lines*

 ```
[...truncated 134 lines...]
skipping: [localhost]

TASK: [Removing possible abandoned dump] ************************************** 
skipping: [localhost]

TASK: [Importing from remote backup] ****************************************** 
skipping: [localhost]

TASK: [Copy latest backup to build folder] ************************************ 
changed: [localhost]

TASK: [Importing from local backup] ******************************************* 
failed: [localhost] => {"failed": true}
msg: ERROR 1114 (HY000) at line 17357: The table 'watchdog' is full


FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=20   changed=13   unreachable=0    failed=1   

Build step 'Execute shell' marked build as failure
```

**Comment by temoor-gilmutdinov** [comment]
test this please

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/50/

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 1]
```diff
@@ -701,3 +701,110 @@ function favrskov_helper_update_7037() {
 function favrskov_helper_update_7038() {
   features_revert_module('config_user_roles');
 }
+
+/**
+ * Need use this update to find all matches.
```
Comment (temoor-gilmutdinov) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: "Need to use this update in order to find all matches."

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 2]
```diff
@@ -701,3 +701,110 @@ function favrskov_helper_update_7037() {
...
+ * Although we found image via uri it's not correct and we lost many of files.
+ * Save file to tmp folder which have error output.
```
Comment (temoor-gilmutdinov) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: "Save file with error output into tmp folder."

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/51/

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 3]
```diff
@@ -701,3 +701,110 @@ function favrskov_helper_update_7037() {
...
+          $last_part = explode('/', $href[1]);
+          $std_file->link = end($last_part);
+        }
+        $files_usage[] = $std_file;
+      }
+    }
+  }
+
+  $location = 'temporary://node_errors.txt';
+  $fp = fopen($location,"w");
```
Comment (sergey347) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: Readability: add space after comma.

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 4]
```diff
@@ -701,3 +701,110 @@ function favrskov_helper_update_7037() {
...
+
+  foreach ($files_usage as $file) {
+    $error = FALSE;
+    if (empty($file->fid) || empty($file->link)) {
+      continue;
+    }
+
+    $fid = db_select('file_managed', 'fm')
+      ->fields('fm', array('fid', 'uri'))
+      ->condition('fid', $file->fid, '=')
```
Comment (sergey347) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: Can reduce usage of '=' as operator due to it`s default value
$query->condition($field, $value = NULL, $operator = '=')

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 5]
```diff
@@ -701,3 +701,110 @@ function favrskov_helper_update_7037() {
...
+  $fields = array_merge($field_long_text, $field_text_summary);
+  $fields_name = array_keys($fields);
+  $result_value = array();
+  foreach ($fields_name as $name) {
+    $data_result = db_select('field_data_' . $name, 'fd')
+      ->fields('fd', array('entity_type', 'entity_id'))
+      ->condition($name . '_value', '%' . db_like('media-image') . '%', 'LIKE');
+    $data_result->addField('fd', $name . '_value', 'value');
+    $data_result = $data_result->execute()->fetchAll();
+    $result_value = array_merge($result_value, array_values($data_result));
```
Comment (temoor-gilmutdinov) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: What is the point of using array_values here?

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 6]
```diff
@@ -701,3 +701,110 @@ function favrskov_helper_update_7037() {
...
+  }
+
+  $files_usage = array();
+  foreach ($result_value as $fields) {
+    if (empty($fields)) {
+      continue;
+    }
+
+    $file_pattern = preg_match_all('/<span.*class="file media-image.*<\/span>/', $fields->value, $file_match);
+    $image_patern = preg_match_all('/<img.*class="media-image.*\/>/', $fields->value, $image_match);
```
Comment (temoor-gilmutdinov) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: Typo in variable name.

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/52/

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install (hunk 4 above)
Comment (temoor-gilmutdinov) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: Equality is default operator for scalar parameters.

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/53/

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 7]
```diff
@@ -701,3 +701,110 @@ function favrskov_helper_update_7037() {
...
+      ->condition('fid', $file->fid, '=')
+      ->execute()
+      ->fetchAll();
+    if (empty($fid[0])) {
+      $output = sprintf('Doesn\'t exist -> Node entity: %s, node id: %s, file id: %s', $file->entity_type, $file->entity_id, $file->fid);
+      $error = TRUE;
+      fwrite($fp, $output . PHP_EOL);
+    }
+
+    if (isset($file->link) && !empty($fid[0])) {
```
Comment (temoor-gilmutdinov) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: You can combine checks
if (empty($fid[0])) {
...
}
elseif (isset($file->link)) {
...
}

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 8]
```diff
@@ -701,3 +701,110 @@ function favrskov_helper_update_7037() {
...
+    if (empty($file->fid) || empty($file->link)) {
+      continue;
+    }
+
+    $fid = db_select('file_managed', 'fm')
+      ->fields('fm', array('fid', 'uri'))
+      ->condition('fid', $file->fid, '=')
+      ->execute()
+      ->fetchAll();
+    if (empty($fid[0])) {
```
Comment (sergey347) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: Consistency in writing if{} block.
Please use one style, i.e. add new line at the begin and end to all if-blocks or do not do it at all.

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/54/

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 9]
```diff
@@ -701,3 +701,110 @@ function favrskov_helper_update_7037() {
...
+
+        if (isset($src[1])) {
+          $last_part = explode('/', $src[1]);
+          $std_file->link = end($last_part);
+        }
+
+        $files_usage[] = $std_file;
+      }
+    }
+    if ($file_match) {
```
Comment (sergey347) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: Seems it is a copy-paste with previous if-block. Merge it.

Comment (dporplenko) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: nope. We have two type of matches for images and for files

Comment (sergey347) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: Image is a subtype of file :)

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/55/

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/56/

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/57/

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/58/

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/59/

**Build Log**
*last 25 lines*

 ```
[...truncated 134 lines...]
skipping: [localhost]

TASK: [Removing possible abandoned dump] ************************************** 
skipping: [localhost]

TASK: [Importing from remote backup] ****************************************** 
skipping: [localhost]

TASK: [Copy latest backup to build folder] ************************************ 
changed: [localhost]

TASK: [Importing from local backup] ******************************************* 
failed: [localhost] => {"failed": true}
msg: ERROR 1114 (HY000) at line 14776: The table 'menu_router' is full


FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=20   changed=13   unreachable=0    failed=1   

Build step 'Execute shell' marked build as failure
```

File: drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install [hunk 10]
```diff
@@ -701,3 +701,100 @@ function favrskov_helper_update_7037() {
...
+        $std_file->fid = $url[1];
+        $std_file->entity_id = $fields->entity_id;
+        $std_file->entity_type = $fields->entity_type;
+        preg_match('/src="([^"]+)"/', $match, $src);
+
+        if (isset($src[1])) {
+          $last_part = explode('/', $src[2]);
+          $std_file->link = end($last_part);
+        }
+
```
Comment (temoor-gilmutdinov) [inline drupal/profiles/favrskov/modules/custom/favrskov_helper/favrskov_helper.install]: This line is not necessary. Same accounts for line 767

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/60/

**Comment by sergey347** [comment]
test this please

**Comment by ffw-bot-ua** [comment]
Build comment file: 
--------------
===============

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.cibox03.m1.propeople.com.ua/job/PR_BUILDER/61/

**Pull request merged**